cocotb/handle.py

```diff
@@ -529,33 +528,40 @@ def __getitem__(self, index):
         return self._sub_handles[index]
 
     def __iter__(self):
-        try:
-            if self._range is None:
-                return
-
-            self._log.debug("Iterating with range [%d:%d]", self._range[0], self._range[1])
-            for i in self._range_iter(self._range[0], self._range[1]):
-                try:
-                    result = self[i]
-                    yield result
-                except IndexError:
-                    continue
-        except GeneratorExit:
-            pass
+        if self._range is None:
+            return
 
+        self._log.debug("Iterating with range [%d:%d]" % (self._range[0], self._range[1]))
```

Comment:
This was actually more correct before - it saves formatting the string unless the log is actually printed

Reply:
Yeah, just a byproduct of the revert I didn't see.